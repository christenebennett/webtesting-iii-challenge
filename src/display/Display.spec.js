import React from 'react';
import renderer from 'react-test-renderer'; 
import { render, cleanup } from 'react-testing-library';

import Display from './Display'

describe('<Display />', () => {
  
  // when unlocked or open use the green-led class
  it('matches snapshot', () => {
    const tree = renderer.create(<Display />)
    expect(tree.toJSON()).toMatchSnapshot();
  })
  
  it('renders <Display /> without crashing', () => {
    render(<Display />);
  });

  // defaults to unlocked and open
  it('defaults to unlocked and open', () => {
    const { getByText } = render(<Display />);
    getByText(/unlocked/i)
    getByText(/open/i)
    cleanup();
  })

  // displays 'Closed' if the closed prop is true
  it('displays as closed if the closed prop is true', () => {
    const { getByText } = render(<Display closed={true} />)
    getByText(/closed/i);
    cleanup();
  })

  // displays 'Open' if the closed prop is false
  it('displays as open if closed prop is false', () => {
    const { getByText } = render(<Display closed={false} />)
    getByText(/open/i);
    cleanup();
  })

  // when locked or closed use the red-led class
  it('displays red leds when closed and locked snapshot', () => {
    const tree = renderer.create(<Display closed={true} locked={true}/>)
    expect(tree.toJSON()).toMatchSnapshot();
  })

  // displays 'Locked' if the locked prop is true and 'Unlocked' if otherwise
  it('displays as locked if the locked prop is true', () => {
    const { getByText } = render(<Display closed={true} locked={true} />)
    getByText(/locked/i)
    cleanup();
  })

});