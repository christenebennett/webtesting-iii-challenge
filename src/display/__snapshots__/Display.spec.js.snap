// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> displays red leds when closed and locked snapshot 1`] = `
<div
  className="display panel"
>
  <div
    className="led red-led"
  >
    Locked
  </div>
  <div
    className="led red-led"
  >
    Closed
  </div>
</div>
`;

exports[`<Display /> matches snapshot 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;
